packages/adapter/lib/adapter/adapter.js

```diff
@@ -358,45 +363,65 @@ function Adapter(options) {
-            } else if (fs.existsSync(this.adapterDir + '/node_modules/' + appName + '.js-controller/node_modules/' + appName + '.' + options.name)) {
-                this.adapterDir += '/node_modules/' + appName + '.js-controller/node_modules/' + appName + '.' + options.name;
-            } else if (fs.existsSync(this.adapterDir + '/node_modules/' + appName + '.js-controller/adapter/' + options.name)) {
-                this.adapterDir += '/node_modules/' + appName + '.js-controller/adapter/' + options.name;
-            } else if (fs.existsSync(this.adapterDir + '/node_modules/' + tools.appName + '.js-controller/node_modules/' + appName + '.' + options.name)) {
-                this.adapterDir += '/node_modules/' + tools.appName + '.js-controller/node_modules/' + appName + '.' + options.name;
-            } else {
-                logger.error(this.namespaceLog + ' Cannot find directory of adapter ' + options.name);
-                this.terminate(EXIT_CODES.CANNOT_FIND_ADAPTER_DIR);
-            }
+        const appName = tools.appName.toLowerCase();
+
+        if (fs.existsSync(controllerDir + '/node_modules/' + appName + '.' + options.name)) {
+            // js-controllers node_modules folder
+            this.adapterDir = controllerDir + '/node_modules/' + appName + '.' + options.name;
+        } else if (fs.existsSync(controllerDir + '/../' + appName + '.' + options.name)) {
+            // same hierarchy as js-controller
+            this.adapterDir = controllerDir + '/../' + appName + '.' + options.name;
+        } else if (fs.existsSync(controllerDir + '/../../node_modules/' + appName + '.' + options.name)) {
+            // testing environment - root folder of monorepo
+            this.adapterDir = controllerDir + '/../../node_modules/' + appName + '.' + options.name;
         } else {
-            this.adapterDir = __dirname.replace(/\\/g, '/');
-            // remove "/lib"
-            this.adapterDir = this.adapterDir.substring(0, this.adapterDir.length - 4);
-            if (fs.existsSync(this.adapterDir + '/node_modules/' + tools.appName + '.' + options.name)) {
-                this.adapterDir += '/node_modules/' + tools.appName + '.' + options.name;
-            } else if (fs.existsSync(this.adapterDir + '/../node_modules/' + tools.appName + '.' + options.name)) {
-                const parts = this.adapterDir.split('/');
-                parts.pop();
-                this.adapterDir = parts.join('/') + '/node_modules/' + tools.appName + '.' + options.name;
-            } else {
-                logger.error(this.namespaceLog + ' Cannot find directory of adapter ' + options.name);
-                this.terminate(EXIT_CODES.CANNOT_FIND_ADAPTER_DIR);
-            }
+            logger.error(this.namespaceLog + ' Cannot find directory of adapter ' + options.name);
+            this.terminate(EXIT_CODES.CANNOT_FIND_ADAPTER_DIR);
+        }
+    }
```

Comment:
Das sollten wir als nächsts auf `require.resolve` umstellen

Reply:
theres already a tools.js method which uses require resolve for this. I've replaced it by the method now